File: src/Moryx.Maintenance.Web/src/modules/api/requests/SaveConfigRequest.ts
/*
 * Copyright (c) 2020, Phoenix Contact GmbH & Co. KG
 * Licensed under the Apache License, Version 2.0
*/

import Config from "../../models/Config";
import { ConfigUpdateMode } from "../../models/ConfigUpdateMode";

export default class SaveConfigRequest {
    public Config: Config;
    public UpdateMode: ConfigUpdateMode;
}

File: src/Moryx.Runtime.Endpoints/Modules/Request/SaveConfigRequest.cs
// Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using Moryx.Runtime.Endpoints.Modules.Models;

namespace Moryx.Runtime.Endpoints.Modules.Request;

/// <summary>
/// Request to save the config of a module
/// </summary>
public class SaveConfigRequest
{
    /// <summary>
    /// Config of the module
    /// </summary>
    public Config Config { get; set; }

    /// <summary>
    /// Update mode how the config will be applied
    /// </summary>
    public ConfigUpdateMode UpdateMode { get; set; }
}